devices.js

```diff
@@ -1395,6 +1395,15 @@ const devices = [
             await configureReporting.deviceTemperature(endpoint);
         },
     },
+    {
+        zigbeeModel: ['lumi.switch.l0agl1'],
+        model: 'SSM-U02',
+        vendor: 'Xiaomi',
+        description: 'Aqara single switch module T1 (without neutral). Doesn\'t work as a router and doesn\'t support power meter',
+        fromZigbee: [fz.on_off],
```

Comment:
Try removing `fromZigbee`, `exposes` and `toZigbee` here and add `extend: generic.switch,`

Reply:
Did it and absolutely nothing has changed: the same error in logs and the device is still working.

Reply:
I'm quite sure the error is not in the config. Probably you are using some incompatible files (e.g. wrong version of zigbee-herdsman-converters with zigbee2mqtt). Try to fully sync with the dev branch and the error should be gone. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

Reply:
Indeed, it works like a charm on dev branch. Thank you!

Reply:
Hi! I have such a relay. After your edits, everything worked except for passing the state. If I turn off the relay with a physical key, then the state in HA remains unchanged.

Reply:
Checked it, and yes - my setup works the same way.
@Koenkk, do you have any idea on how to make it "see" the state changes when pressing the physical key?

Reply:
Try changing the entry to:

```js
{
    zigbeeModel: ['lumi.switch.l0agl1'],
    model: 'SSM-U02',
    vendor: 'Xiaomi',
    description: 'Aqara single switch module T1 (without neutral). Doesn\'t work as a router and doesn\'t support power meter',
    extend: preset.switch,
    meta: {configureKey: 1},
    configure: async (device, coordinatorEndpoint) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff']);
        await reporting.onOff(endpoint);
    },
},
```

Reply:
I couldn't make it work with this config, though my test were not 100% correct probably, since I'm running z2m in docker on Synology. I tried:
- just altering the `devices.js` inside the container, then restart it --> container crashes
- doing as described [here](https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html#docker) --> Synology lacks the npm's dependencies
- the same as above, but building the stuff in official node docker container (tried buster and alpine), then moving the entire `zigbee-herdsman-converters` folder to the z2m container (as volume, of course) --> container crashes
- something similar I even can't remember but still - container crashes

@Koenkk what am I doing wrong?
@opa4h could you try the proposed solution?

Reply:
Checked it out! Now it works!

Reply:
Great news! I’ll make a PR with this change.